client/allocrunner/taskrunner/getter/params.go

```diff
@@ -0,0 +1,159 @@
+package getter
+
+import (
+	"context"
+	"encoding/json"
+	"io"
+	"io/fs"
+	"strings"
+	"time"
+
+	"github.com/hashicorp/go-cleanhttp"
+	"github.com/hashicorp/go-getter"
+	"github.com/hashicorp/nomad/helper"
+	"golang.org/x/exp/maps"
+	"golang.org/x/exp/slices"
+)
+
+// parameters is encoded by the Nomad client and decoded by the getter sub-process
+// so it can know what to do. We use standard IO instead of parameters variables
+// because the job-submitter has control over the parameters and that is scary,
+// see https://www.opencve.io/cve/CVE-2022-41716.
+type parameters struct {
+	// Config
+	HTTPReadTimeout time.Duration `json:"http_read_timeout"`
+	HTTPMaxBytes    int64         `json:"http_max_bytes"`
+	GCSTimeout      time.Duration `json:"gcs_timeout"`
+	GitTimeout      time.Duration `json:"git_timeout"`
+	HgTimeout       time.Duration `json:"hg_timeout"`
+	S3Timeout       time.Duration `json:"s3_timeout"`
+
+	// Artifact
+	Mode        getter.ClientMode   `json:"artifact_mode"`
+	Source      string              `json:"artifact_source"`
+	Destination string              `json:"artifact_destination"`
+	Headers     map[string][]string `json:"artifact_headers"`
+
+	// Task Environment
+	TaskDir string `json:"task_dir"`
+}
+
+func (p *parameters) reader() io.Reader {
+	b, err := json.Marshal(p)
+	if err != nil {
+		b = nil
+	}
+	return strings.NewReader(string(b))
+}
+
+func (p *parameters) read(r io.Reader) error {
+	return json.NewDecoder(r).Decode(p)
+}
+
+// deadline returns an absolute deadline before the artifact download
+// sub-process forcefully terminates. The default is 1 hour, unless
+// one or more getter configurations is set higher.
+func (p *parameters) deadline() time.Duration {
+	const minimum = 1 * time.Hour
```

Comment:
Not a big deal but why so much longer than our default value for the getter configurations (30m)?

Reply:
Good point - the idea here is there are 3 modes of timeout
- the getter internal timeout
- the sub-process self termination timeout
- the nomad agent watchdog timeout
and we want to give a chance for these to occur in that order.
The "double the internal default timeout" is overkill, we need only a small
grace period over the sub-process self termination timeout.